src/Service/OrbitService.cpp

```diff
@@ -138,15 +138,17 @@ int OrbitService::Run(std::atomic<bool>* exit_requested) {
 
   std::unique_ptr<OrbitGrpcServer> grpc_server = CreateGrpcServer(grpc_port_, dev_mode_);
   if (grpc_server == nullptr) {
-    ORBIT_ERROR("Unable to create gRPC server.");
-    return -1;
+    constexpr std::string_view kErrorMessage = "Unable to create gRPC server.";
+    ORBIT_ERROR("%s", kErrorMessage);
+    return ErrorMessage{std::string{kErrorMessage}};
```

Comment:
`constexpr const char*` and `std::string_view` have basically no difference - except that the constructor of std::string that takes a string_view is explicit, hence we need `std::string{...}`.

`const std::string kErrorMe...` means a string will get heap allocated and since it's const it will get copied into the error message. So another heap allocation takes place. But a non-const string would just work fine - in this case.

I prefer using a constexpr string_view and doing the heap allocation when the error message is constructed because that's the most generic approach. I would also use it when the constant was defined on compilation unit level or in a header. It would always be the same.

Reply:
Thank you!

Reply:
For me to learn as well: so you prefer `std::string_view` over `constexpr const char*` because you want the construction of `std::string` to be explicit, nothing else?

Reply:
The explicit string construction is more of a side effect that I indeed also like.

To me the most important argument is that it is more generic. The `const char*` constructor of `std::string` does not work with binary strings that might contain null characters. But string_view works in all possible situations. I hope I can save myself and others a couple of hours of searching for a bug like this by establishing good practices.

Also the `std::string::string(std::string_view)` constructor is technically faster because it doesn't need to call `std::strlen` which of course shouldn't be the deciding argument.